src/apps/Altinn.AccessManagement/src/Altinn.AccessManagement.Api.Internal/Controllers/Bff/ConsentController.cs

```diff
@@ -0,0 +1,277 @@
+            {
+                bool isAuthorized = await AuthorizeResourceAccess("altinn_access_management", resourePartUuuid, User, "read");
+
+                if (isAuthorized)
+                {
+                    return Ok(consent.Value);
+                }
+            }
+
+            return Unauthorized();
+        }
+
+        /// <summary>
+        /// Endpoint to get the redirect URL for a consent request.
+        /// </summary>
+        [HttpGet]
+        [Route("consentrequests/{requestId}/redirecturl")]
+        public async Task<IActionResult> GetConsentRedirectUrl([FromRoute] Guid requestId, CancellationToken cancellationToken = default)
+        {
+            string redirectUrl = await consentService.GetRequestRedirectUrl(requestId, cancellationToken);
+            if (string.IsNullOrEmpty(redirectUrl))
+            {
+                return NotFound("Redirect URL not found for the consent request.");
+            }
+
+            return Ok(new ConsentRedirectUrlDto() { Url = redirectUrl });
+        }
+
+        /// <summary>
+        /// Endpoint to approve a consent request
+        /// The authenticated user must fullfill the requirements to approve the request.
+        /// - Have right for accessmanagement for the party that is requesting the consent
+        /// - Have the right to delegate the requested rights. Either by having the right self or beeing the main administrator
+        /// </summary>
+        [HttpPost]
+        [Authorize(Policy = AuthzConstants.SCOPE_PORTAL_ENDUSER)]
+        [Route("consentrequests/{requestId}/accept")]
+        public async Task<IActionResult> Accept(Guid requestId, [FromBody] ConsentContextDto context, CancellationToken cancellationToken = default)
+        {
+            Guid? performedBy = UserUtil.GetUserUuid(User);
+            if (performedBy == null)
+            {
+                return Unauthorized();
+            }
+
+            if (context == null)
+            {
+                return BadRequest("Consent context is required");
+            }
+
+            Core.Models.Consent.ConsentPartyUrn performedByParty = Core.Models.Consent.ConsentPartyUrn.PartyUuid.Create(performedBy.Value);
+            Result<ConsentRequestDetails> consentRequest = await _consentService.GetRequest(requestId, performedByParty, true, cancellationToken);
+            if (consentRequest.IsProblem)
+            {
+                return consentRequest.Problem.ToActionResult();
+            }
+
+            // Check if the user is authorized to accept consent request in general. 
+            if (consentRequest.Value.From.IsPartyUuid(out Guid resourePartUuuid))
+            {
+                bool isAuthorized = await AuthorizeResourceAccess("altinn_access_management", resourePartUuuid, User, "write");
+
+                if (!isAuthorized)
+                {
+                    return Unauthorized();
```

Comment:
Forbidden? Er for så vidt poentsielt mange plasser du nå returnerer Unauthorized som kanskje burde vært Forbidden, for å kunne skille på at man ikke har gyldig pålogging/token vs. man forsøker noe man ikke har lov til.

Vet ikke hvordan frontend evt. skal kunne ellers tolke respons, når det heller ikke benyttes noen ProblemDetails respons

Reply:
Endret til forbid